server/athenian/api/experiments/preloading/entries.py

```diff
@@ -73,10 +109,63 @@ class PreloadedReleaseToPullRequestMapper(ReleaseToPullRequestMapper):
     release_loader = PreloadedReleaseLoader
 
 
+class PreloadedOpenPRFactsLoader(OpenPRFactsLoader):
+    """Loader for preloaded open PRs facts."""
+
+    @classmethod
+    @sentry_span
+    async def load_open_pull_request_facts_unfresh(cls,
+                                                   prs: Iterable[str],
+                                                   time_from: datetime,
+                                                   time_to: datetime,
+                                                   exclude_inactive: bool,
+                                                   authors: Mapping[str, str],
+                                                   account: int,
+                                                   pdb: databases.Database,
+                                                   ) -> Dict[str, PullRequestFacts]:
+        """
+        Fetch preloaded precomputed facts about the open PRs from the DataFrame.
+
+        We don't filter PRs by the last update here.
+
+        :param authors: Map from PR node IDs to their author logins.
+        :return: Map from PR node IDs to their facts.
+        """
+        cached_df = pdb.cache.dfs[PCID.done_pr_facts]
+        df = cached_df.df
+        mask = (
+            (df["acc_id"] == account)
+            & df["pr_node_id"].isin(prs)
+        )
+
+        if exclude_inactive:
+            date_range = build_days_range(time_from, time_to)
+            mask &= df["activity_days"].apply(lambda v: bool(set(v).intersection(date_range)))
+
+        open_prs_facts = cached_df.filter(mask, columns=[
+            "pr_node_id", "repository_full_name", "data"])
+        if open_prs_facts.empty:
+            return {}
+
+        facts = {}
+        for pr_node_id, repository_full_name, data in zip(
+                open_prs_facts["pr_node_id"].values,
+                open_prs_facts["repository_full_name"].values,
+                open_prs_facts["data"].values,
+        ):
+            node_id = pr_node_id.rstrip()
+            facts[node_id] = PullRequestFacts(
+                data=data,
+                repository_full_name=repository_full_name,
+                author=authors[node_id])
```

Comment:
In that case,
```suggestion
        facts = {
            facts[pr_node_id] = PullRequestFacts(
                data=data,
                repository_full_name=repository_full_name,
                author=authors[pr_node_id])
            for pr_node_id, repository_full_name, data in zip(
                open_prs_facts["pr_node_id"].str.rstrip().values,
                open_prs_facts["repository_full_name"].values,
                open_prs_facts["data"].values,
            )
        }        
```

If we didn't have `str.rstrip()` in Pandas, I would propose this:
```python
        facts = {
            facts[node_id := pr_node_id.rstrip()] = PullRequestFacts(
                data=data,
                repository_full_name=repository_full_name,
                author=authors[node_id])
            for pr_node_id, repository_full_name, data in zip(
                open_prs_facts["pr_node_id"].values,
                open_prs_facts["repository_full_name"].values,
                open_prs_facts["data"].values,
            )
        }  
```